src/seedu/addressbook/data/person/Email.java

```diff
@@ -6,15 +6,14 @@
  * Represents a Person's email in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidEmail(String)}
  */
-public class Email {
+public class Email extends Contact {
```

Comment:
If you push the constructor to the superclass, you need to call `super()` method in the `Email` constructor (basically all child classes' constructors). The `super()` method is used to call superclass constructor.

So something like

```java
public Email(String email, boolean isPrivate) throws IllegalValueException {
        super(email, isPrivate, EMAIL_VALIDATION_REGEX, MESSAGE_EMAIL_CONSTRAINTS);
}
```